src/GraphQL.Tests/Bugs/Bug_nested_type_names_dont_conflict.cs

```diff
@@ -0,0 +1,71 @@
+using GraphQL.SystemTextJson;
+using GraphQL.Types;
+using Xunit;
+
+namespace GraphQL.Tests.Bugs
+{
+    public class Bug_nested_type_names_dont_conflict :
+        QueryTestBase<Bug_nested_type_names_dont_conflict.MutationSchema>
+    {
+        [Fact]
+        public void nested_type_names_dont_conflict()
+        {
+            var inputs = @"{ ""input_0"": { ""id"": ""123""} }".ToInputs();
+            var query = @"
+mutation M($input_0: MyInput!) {
+  run(input: $input_0)
+}
+";
+            var expected = @"{ ""run"": ""123"" }";
+            AssertQuerySuccess(query, expected, inputs);
+        }
+
+        public class MutationSchema : Schema
+        {
+            public MutationSchema()
+            {
+                Mutation = new MyMutation();
+            }
+        }
+
+        public class MyMutation : ObjectGraphType
+        {
+            public MyMutation()
+            {
+                Field<StringGraphType>(
+                    "run",
+                    arguments: new QueryArguments(new QueryArgument<MyInputClass.MyInput> {Name = "input"}),
+                    resolve: ctx => ctx.GetArgument<MyInputClass>("input").Id);
+                Field<StringGraphType>(
+                    "run2",
+                    arguments: new QueryArguments(new QueryArgument<MyInputClass2.MyInput> { Name = "input" }),
+                    resolve: ctx => ctx.GetArgument<MyInputClass2>("input").Id);
+            }
+        }
+
+        public class MyInputClass
+        {
+            public string Id { get; set; }
+
+            public class MyInput : InputObjectGraphType
+            {
+                public MyInput()
+                {
+                    Field<NonNullGraphType<StringGraphType>>("id");
+                }
+            }
+        }
+        public class MyInputClass2
+        {
+            public string Id { get; set; }
+
+            public class MyInput : InputObjectGraphType
+            {
+                public MyInput()
+                {
+                    Field<NonNullGraphType<StringGraphType>>("id");
```

Comment:
```suggestion
                    Name = "MyInput2";
                    Field<NonNullGraphType<StringGraphType>>("id");
```
Is this not satisfactory?

Reply:
shouldnt this be supported OOTB?

Reply:
OK

Reply:
In general, the problem is wider and inherent to all grapj types. By default, the type name is defined in `GraphType.cs` as:
```
  private string GetDefaultName()
        {
            var type = GetType();

            var attr = type.GetCustomAttribute<GraphQLMetadataAttribute>();

            if (!string.IsNullOrEmpty(attr?.Name))
            {
                return attr.Name;
            }

            var name = type.Name.Replace('`', '_');
            if (name.EndsWith(nameof(GraphType), StringComparison.InvariantCulture))
                name = name.Substring(0, name.Length - nameof(GraphType).Length);

            return name;
        }
```

That is, only the name type is used without namespace and without parent types. This is done intentionally for ease of names. We _can_ change the algorithm to take into account paren types and namespaces, but it will be a breaking change. I suggest to use `GlobalSwitches`.

Reply:
I'll post a PR.